test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs

```diff
@@ -677,6 +677,40 @@ public class RestoreCommandTests
             }
         }
 
+        [Fact] 
+        public async Task RestoreCommand_InstallPackageWithManyDependencies()
+        {
+            // Arrange
+            var sources = new List<PackageSource>();
+            sources.Add(new PackageSource("https://www.myget.org/F/aspnetcidev/api/v3/index.json"));
```

Comment:
Thinking about this more, it would probably be better to just use local packages. Once the simple package test utility is added from my PR it should be trivial to make a package with 500 dependencies and restore it.